cmd/worker/main.go

```diff
@@ -185,11 +194,23 @@ func main() {
 		log.Label("package_bucket", packagesBucket),
 		log.Label("results_bucket", resultsBucket),
 		log.Label("file_write_results_bucket", fileWritesBucket),
-		log.Label("image_tag", imageTag))
+		log.Label("image_tag", imageTag),
+		log.Label("topic_notification", notificationTopicURL))
 
 	for {
-		err := messageLoop(ctx, subURL, packagesBucket, resultsBucket, fileWritesBucket, imageTag)
-		if err != nil {
+		// default values set so the notification topic is optional
+		// if the environment variable is unable to be retrieved,
+		// no notifications are sent and we continue without notifying
+		var err error = nil
+		var notificationTopic *pubsub.Topic = nil
+		if notificationTopicURL != "" {
+			notificationTopic, err = pubsub.OpenTopic(ctx, notificationTopicURL)
+			defer notificationTopic.Shutdown(ctx)
+		}
+		if err == nil {
```

Comment:
This logic is getting a bit too complicated.

Let's also move the `pubsub.OpenTopic` topic bit into the `messageLoop`. This is consistent with the `pubsub.OpenSubscription` in that function, and also does the same thing re retrying whenever an error is encountered.